crates/noirc_evaluator/src/lib.rs

```diff
@@ -40,33 +40,43 @@ pub fn create_circuit(
     is_blackbox_supported: IsBlackBoxSupported,
     enable_logging: bool,
     show_output: bool,
-) -> Result<Circuit, RuntimeError> {
+) -> Result<(Circuit, BTreeMap<String, Vec<Witness>>), RuntimeError> {
     let mut evaluator = Evaluator::new();
 
     // First evaluate the main function
-    evaluator.evaluate_main_alt(program, enable_logging, show_output)?;
+    evaluator.evaluate_main_alt(program.clone(), enable_logging, show_output)?;
+
+    // TODO: shrink this ungodly mess.
```

Comment:
Noting TODO -- though its not _that_ _that_ bad

Reply:
Addressed in [c08d154](https://github.com/noir-lang/noir/pull/851/commits/c08d154e20cc9e40f3cce336f773f22e37b23b21)